src/m365/spo/commands/file/file-rename.ts

```diff
@@ -0,0 +1,138 @@
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  sourceUrl: string;
+  targetFileName: string;
+  force?: boolean;
+}
+
+class SpoFileRenameCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FILE_RENAME;
+  }
+
+  public get description(): string {
+    return 'Renames a file';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.force = !!args.options.force;
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const webUrl = args.options.webUrl;
+    const originalFileServerRelativeUrl: string = urlUtil.getServerRelativePath(args.options.webUrl, args.options.sourceUrl);
+    
+    this
+      .fileExists(originalFileServerRelativeUrl, webUrl)
+      .then(_ => {
+        if (args.options.force) {
+          return this.deleteTargetItem(webUrl, args.options.sourceUrl, args.options.targetFileName);
+        } 
+        return Promise.resolve();
+      })
+      .then(_ => {
+        const requestBody: any = {
+          formValues : [{
+            FieldName: 'FileLeafRef',
+            FieldValue: args.options.targetFileName
+          }]
+        };
+
+        const requestOptions: any = {
+          url: `${webUrl}/_api/web/GetFileByServerRelativeUrl('${encodeURIComponent(originalFileServerRelativeUrl)}')/ListItemAllFields/ValidateUpdateListItem()`,
+          headers: {
+            'accept': 'application/json;odata=nometadata'
+          },
+          data: requestBody,
+          responseType: 'json'
+        };
+
+        return request.post(requestOptions);
+      })
+      .then((resp: any): Promise<any> => {
+        return new Promise<void>((resolve: () => void): void => {
+          logger.log(resp.value);
+          resolve();
+        });
+      })
+      .then(_ => cb(), (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private fileExists(originalFileServerRelativeUrl: string, webUrl: string): Promise<void> {
+    const requestUrl = `${webUrl}/_api/web/GetFileByServerRelativeUrl('${encodeURIComponent(originalFileServerRelativeUrl)}')`;
+    const requestOptions: any = {
+      url: requestUrl,
+      method: 'GET',
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+    return request.get(requestOptions);
+  }
+
+  private deleteTargetItem(webUrl: string, sourceUrl: string, targetFileName: string): Promise<void> {
+    const targetFileServerRelativeUrl: string = `${urlUtil.getServerRelativePath(webUrl,sourceUrl.substring(0, sourceUrl.lastIndexOf('/')))}/${targetFileName}`;
+
+    const options: SpoFileRemoveOptions = {
+      webUrl: webUrl,
+      url: targetFileServerRelativeUrl,
+      recycle: true,
+      confirm: true
+    };
```

Comment:
We should pass the `verbose` and `debug` options here to include the output of this command in the logs

Reply:
@MathijsVerbeeck I was going over the last review before I start a new one and this comments is marked as resolved but TBH I cannot see the that it is done 😉. I would expect that 
```
debug: this.debug,
verbose: this.verbose
```
are added to the options object 🤔. Before I move forward could we double check this one 🙏. Maybe it was just overlooked or it is done but I just don't see it 👍